### PR TITLE
feat: add jwt authn to SecurityPolicy

### PR DESCRIPTION
**Pull request by zhaohuabing**
**What type of PR is this?**
This PR adds JWT Authentication to the `SecurityPolicy`.

Related issue: https://github.com/envoyproxy/gateway/issues/1845

This feature is intended to replace the JWT Authn configuration in the existing `AuthenticationFilter`.  `AuthenticationFilter` will be removed in a follow-up PR.

### CODE REVIEW

File: api/v1alpha1/securitypolicy_types.go [hunk 1]
```diff
@@ -43,17 +43,25 @@ type SecurityPolicySpec struct {
...
 	// CORS defines the configuration for Cross-Origin Resource Sharing (CORS).
+	//
+	// +optional
 	CORS *CORS `json:"cors,omitempty"`
+
+	// JWTAuthentication defines the configuration for JSON Web Token (JWT)
+	// authentication.
+	//
+	// +optional
+	JWTAuthentication *JWTAuthentication `json:"jwtAuthentication,omitempty"`
```
Comment (zhaohuabing) [inline api/v1alpha1/securitypolicy_types.go:54]: We have two choices for the naming here:
* `JWT`, shorter, but a bit vague
* `JWTAuthentiation`, clear, but a bit lenghty

Comment (arkodg) [reply api/v1alpha1/securitypolicy_types.go:54]: I vote for `jwt`

Comment (zirain) [reply api/v1alpha1/securitypolicy_types.go:54]: +1 for `jwt`

Comment (zhaohuabing) [reply api/v1alpha1/securitypolicy_types.go:54]: JWT then

Comment (arkodg) [reply api/v1alpha1/securitypolicy_types.go:54]: `jwt` can be used for `auth`, for converting claim to header, maybe also be used for authz of claim

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zirain — APPROVED** [review]

**Pull request merged**